docs/sequence.rst

```diff
@@ -0,0 +1,647 @@
+    > "base"
+
+    (Seq.fromList ["base", "containers"]) Seq.!? 2
+    > Nothing
+
+    Seq.index (Seq.fromList ["base", "containers"]) 2
+    > "*** Exception: index out of bounds
+
+
+Modification
+^^^^^^^^^^^^
+
+Inserting an element
+""""""""""""""""""""
+
+::
+
+    Seq.insertAt :: Int -> a -> Seq a -> Seq a
+    Seq.insertAt i x xs = ...
+
+:seq:`insertAt` inserts ``x`` into ``xs`` at the index ``i``, shifting the rest
+of the sequence over.
+
+::
+
+    Seq.insertAt 0 "idris" (Seq.fromList ["haskell", "rust"])
+    > fromList ["idris","haskell","rust"]
+
+See also `Adding to an existing sequence`_.
+
+Replace an element
+""""""""""""""""""
+
+::
+
+    Seq.update :: Int -> a -> Seq a -> Seq a
+    Seq.update i x xs = ...
+
+:seq:`update` replaces the element at position ``i`` in the sequence with
+``x``. If the index is out of bounds then the original sequence is returned.
+
+::
+
+    Seq.update 0 "hello" (Seq.fromList ["hi", "world", "!"])
+    > fromList ["hello","world","!"]
+
+    Seq.update 3 "OUTOFBOUNDS" (Seq.fromList ["hi", "world", "!"])
+    > fromList ["hi","world","!"]
+
+Adjust/modify an element
+""""""""""""""""""""""""
+
+*Since version 0.5.8*
+
+::
+
+    adjust' :: forall a. (a -> a) -> Int -> Seq a -> Seq a
+    adjust' f i xs = ...
+
+`adjust'
+<https://hackage.haskell.org/package/containers-0.5.10.2/docs/Data-Sequence.html#v:adjust-39->`_
+updates the element at position ``i`` in the sequence by applying the function
+``f`` to the existing element. If the index is out of bounds then the original
+sequence is returned.
+
+::
+
+    Seq.adjust' (*10) 0 (Seq.fromList [1, 2, 3])
+    > fromList [10,2,3]
+
+    Seq.adjust' (*10) 3 (Seq.fromList [1, 2, 3])
+    > fromList [1,2,3]
+
+.. NOTE::
+   If you're using an older version of containers which only has ``adjust``, be
+   careful because it can lead to `poor performance and space leaks
+   <https://hackage.haskell.org/package/containers-0.5.10.2/docs/Data-Sequence.html#v:adjust>`_.
+
+Modifying all elements
+""""""""""""""""""""""
+
+::
+
+    fmap :: (a -> b) -> Seq a -> Seq b
+    fmap f xs = ...
+
+    Seq.mapWithIndex :: (Int -> a -> b) -> Seq a -> Seq b
+    Seq.mapWithIndex f xs = ...
+
+:seq:`fmap` transform each element of the sequence with the function
+``f``. ``fmap`` is provided by the `Functor
+<https://wiki.haskell.org/Typeclassopedia#Functor>`_ instance for sequences and
+can also be written infix using the ``<$>`` operator.
+
+:seq:`mapWithIndex` allows you to do a similar transformation but given you the
+index that each element is at.
+
+::
+
+    fmap (*10) (Seq.fromList [1, 2, 3])
+    -- = fromList [1*10, 2*10, 3*10]
+    > fromList [10,20,30]
+
+    (*10) <$> Seq.fromList [1, 2, 3]
+    -- = fromList [1*10, 2*10, 3*10]
+    > fromList [10,20,30]
+
+    let myMapFunc index val = index * val
+
+    Seq.mapWithIndex myMapFunc (Seq.fromList [1, 2, 3])
+    -- = fromList [0*1, 1*2, 2*3]
+    > fromList [0,2,6]
+
+
+Delete an element
+"""""""""""""""""
```

Comment:
It seems to me that this should be adjacent to `insertAt`, but maybe that's wrong?

Reply:
Agreed, moved.